### PR TITLE
Add a simple prompt for port choices

### PR DESCRIPTION
**Pull request by alisomay**
I have created a working prompt for choosing ports. 
It was actually for fun. I think it can be done a bit better.
Please check it out when you have time. 
Best!

### CODE REVIEW

**Review by alexrodi — APPROVED** [review]

File: src/midi.rs [hunk 1]
```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
...
+    if let None = re_ask {
+        println!("Please choose your midi input.");
+    }
+    match read_input().expect("Couldn't read input.").parse::<usize>() {
+        Ok(c) => {
+            if c < ports.len() {
+                println!("You have chosen {}", c);
+                choice = c;
+            } else {
+                println!("Please enter a valid number of a port.");
```
Comment (alexrodi) [inline src/midi.rs]:
```suggestion
                println!("{}: Not a valid port number", c);
```

Comment (alexrodi) [reply src/midi.rs]: Let's keep it light, we can cut the "please".
Sounds too .... microsoft excel 😄

File: src/midi.rs [hunk 2]
```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
...
     port: MidiInputPort,
     port_name: String,
-    midi_in: Option<MidiInput>
+    midi_in: Option<MidiInput>,
+}
+
+fn port_prompt(ports: &Vec<MidiInputPort>, re_ask: Option<()>) -> usize {
+    let mut choice: usize = 0;
+    if let None = re_ask {
+        println!("Please choose your midi input.");
```
Comment (alexrodi) [inline src/midi.rs]:
```suggestion
        println!("Choose your midi input.");
```
I really like the usage of "your", makes it feel less machiney :)

File: src/midi.rs [hunk 3]
```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
...
+            if c < ports.len() {
+                println!("You have chosen {}", c);
+                choice = c;
+            } else {
+                println!("Please enter a valid number of a port.");
+                port_prompt(ports, Some(()));
+            }
+        }
+        Err(_) => {
+            println!("Please enter a number of the port you want to choose instead.");
```
Comment (alexrodi) [inline src/midi.rs]:
```suggestion
            println!("{}: Not a valid port number", c);
```
I would go with the exact same message, keeps it simple.
If they're different, you get a different error message for `-1` and `99` which seems wonky.. :)

**Review by alexrodi — APPROVED** [review]
Approved with suggested changes

File: src/midi.rs (hunk 3 above)
Comment (alisomay) [reply src/midi.rs]: You might be right I agree :)

File: src/midi.rs (hunk 2 above)
Comment (alisomay) [reply src/midi.rs]: :D

File: src/midi.rs (hunk 1 above)
Comment (alisomay) [reply src/midi.rs]: sdadfgdsaf

File: src/midi.rs [hunk 4]
```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
...
+
+fn port_prompt(ports: &Vec<MidiInputPort>, re_ask: Option<()>) -> usize {
+    let mut choice: usize = 0;
+    if let None = re_ask {
+        println!("Please choose your midi input.");
+    }
+    match read_input().expect("Couldn't read input.").parse::<usize>() {
+        Ok(c) => {
+            if c < ports.len() {
+                println!("You have chosen {}", c);
```
Comment (alexrodi) [inline src/midi.rs]:
```suggestion
 
```
I think this print is unnecessary, a print with the port name will be displayed anyways

File: src/midi.rs [hunk 5]
```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
...
+
         let ports = get_ports(&midi_in, true)?;
-    
+        let port = port_prompt(&ports, None);
+
         let port = ports.get(port).ok_or("port not found")?.clone();
         let port_name = midi_in.port_name(&port)?;
-    
+
         println!("Selecting: {}", port_name);
```
Comment (alexrodi) [inline src/midi.rs]:
```suggestion
        println!("Selected: {}", port_name);
```
Using the infinitive makes you question if the action is done or still executing

**Pull request merged**